core/__tests__/tasks/appRefreshQuery/query.ts

```diff
@@ -0,0 +1,154 @@
+import { helper } from "@grouparoo/spec-helper";
+import { api, task, specHelper } from "actionhero";
+import {
+  Source,
+  Schedule,
+  App,
+  Run,
+  plugin,
+  GrouparooModel,
+  AppRefreshQuery,
+} from "../../../src";
+import { RunOps } from "../../../src/modules/ops/runs";
+
+describe("tasks/appRefreshQuery:query", () => {
+  let model: GrouparooModel;
+  let source: Source;
+  let schedule: Schedule;
+
+  helper.grouparooTestServer({ truncate: true, enableTestPlugin: true });
+  beforeEach(async () => await api.resque.queue.connection.redis.flushdb());
+
+  beforeAll(async () => {});
+
+  describe("appRefreshQuery:query", () => {
+    let model: GrouparooModel;
+    let app: App;
+    let source: Source;
+    let schedule: Schedule;
+    let appRefreshQuery: AppRefreshQuery;
+
+    beforeAll(async () => {
+      ({ model } = await helper.factories.properties());
+      app = await helper.factories.app(model);
+      source = await helper.factories.source(app);
+      await source.setOptions({ table: "test table" });
+      await source.setMapping({ id: "userId" });
+      await source.update({ state: "ready" });
+
+      schedule = await helper.factories.schedule(source);
+      await schedule.update({ state: "ready" });
+
+      appRefreshQuery = new AppRefreshQuery({
+        appId: app.id,
+        refreshQuery: "SELECT 'hi' AS name;",
+        state: "ready",
+      });
+      await appRefreshQuery.save();
+    });
+
+    test("can be enqueued and run", async () => {
+      await task.enqueue("appRefreshQuery:query", {
+        appRefreshQueryId: appRefreshQuery.id,
+      }); //does not throw
+
+      await specHelper.runTask("appRefreshQuery:query", {
+        appRefreshQueryId: appRefreshQuery.id,
+      });
+
+      const enqueuedRuns = await Run.findAll({
+        where: { creatorId: schedule.id, state: "running" },
+      });
+      expect(enqueuedRuns.length).toBe(1);
+
+      Run.truncate();
```

Comment:
[suggestion] - you might want to move `Run.truncate()` into a beforeAll or afterAll to ensure the tests always start in a cleaned-up state

Reply:
Do you mean a beforeEach since I'm calling it at the end of multiple tests?

Reply:
oops! yep - `beforeEach`